drivers/base/bootvid/common.c

```diff
@@ -15,8 +15,63 @@ ULONG VidpScrollRegion[4] =
     SCREEN_HEIGHT - 1
 };
 
+/*
+ * Boot video driver default palette is similar to the standard 16-color
+ * CGA palette, but it has Red and Blue channels swapped, and also dark
+ * and light gray colors swapped.
+ */
+const PALETTE_ENTRY DefaultPalette[BV_MAX_COLORS] =
+{
+    {0x00, 0x00, 0x00}, /* Black */
+    {0x80, 0x00, 0x00}, /* Red */
+    {0x00, 0x80, 0x00}, /* Green */
+    {0x80, 0x80, 0x00}, /* Brown */
+    {0x00, 0x00, 0x80}, /* Blue */
+    {0x80, 0x00, 0x80}, /* Magenta */
+    {0x00, 0x80, 0x80}, /* Cyan */
+    {0x80, 0x80, 0x80}, /* Dark Gray */
+    {0xC0, 0xC0, 0xC0}, /* Light Gray */
+    {0xFF, 0x00, 0x00}, /* Light Red */
+    {0x00, 0xFF, 0x00}, /* Light Green */
+    {0xFF, 0xFF, 0x00}, /* Yellow */
+    {0x00, 0x00, 0xFF}, /* Light Blue */
+    {0xFF, 0x00, 0xFF}, /* Light Magenta */
+    {0x00, 0xFF, 0xFF}, /* Light Cyan */
+    {0xFF, 0xFF, 0xFF}, /* White */
+};
+
+static BOOLEAN ClearRow = FALSE;
+
 /* PRIVATE FUNCTIONS **********************************************************/
 
+VOID
+NTAPI
+InitializePalette(VOID)
+{
+    ULONG i, pal;
+
+    for (i = 0; i < BV_MAX_COLORS; i++)
+    {
+        pal = (DefaultPalette[i].Red << 16) | (DefaultPalette[i].Green << 8) | DefaultPalette[i].Blue;
+        SetPaletteEntry(i, pal);
+    }
+}
+
+VOID
+NTAPI
+InitPaletteWithTable(
+    _In_ PULONG Table,
+    _In_ ULONG Count)
+{
+    ULONG i;
+    PULONG Entry = Table;
+
+    for (i = 0; i < Count; i++, Entry++)
```

Comment:
Please move `InitPaletteWithTable` back to the VGA code because the `VideoSync()` should be called before starting the loop (PC-98 version).

Reply:
What "VideoSync" you're talking about? (also I've checked in PC98's bootvid of Win2k and there is no such thing, both InitPaletteWithTable and InitializePalette are the same as in regular PC vga, except that: SetPaletteEntry implementation changes (of course), and also that InitializePalette sets all suppemental entries indices 16 to 255 to black.)

Reply:
VideoSync is function that waits for the vertical sync before setting hardware palette registers.

Reply:
But then shouldn't it be called when setting each palette entry?